classes/BadpwFailedLoginsDAO.inc.php

```diff
@@ -61,19 +61,19 @@ public function getByUsername(string $username) : ?BadpwFailedLogins {
 			SELECT *
 			FROM badpw_failedlogins
 			WHERE username = ?
-		', $username);
-		$row = $result->GetRowAssoc(false);
-		if ($result->RowCount()) {
-			return new BadpwFailedLogins($row['username'], $row['count'], strtotime($row['failed_login_time']));
-		}
-		/** @var UserDao */
-		$userDao = DAORegistry::getDAO('UserDAO');
-		$user = $userDao->getByUsername($username);
-		if ($user) {
-			$badpwObj = new BadpwFailedLogins($username, 0, time());
-			$this->_insertObject($badpwObj);
-			return $badpwObj;
+		', [(string)$username]);
```

Comment:
The `$username` parameter is defined as a string, so the cast here shouldn't be needed.